source/common/http/user_agent.cc

```diff
@@ -12,11 +12,11 @@ namespace Envoy {
 namespace Http {
 
 void UserAgent::completeConnectionLength(Stats::Timespan& span) {
-  if (!stats_) {
+  if (!stats_ || !scope_) {
```

Comment:
nit: I think scope_ must be not nullptr if stats is not nullptr?

Reply:
Yup, I was being a little conservative in case the implementation changes below, but I'm fine leaving it out if the `if`.

Reply:
Removed it and added a quick comment noting the expectation.